### PR TITLE
feat!: add CreateInteractionResponse struct

### PR DESCRIPTION
**Pull request by Acnologla**
closes [#469](https://github.com/andersfylling/disgord/issues/469)

Example: 
```go
package main

import (
    "context"
    "os"

    "github.com/andersfylling/disgord"
)

func interaction(session disgord.Session, evt *disgord.InteractionCreate) {
    if evt.Type == disgord.InteractionApplicationCommand {
        f1, err := os.Open("myfavouriteimage.jpg")
        if err != nil {
            panic(err)
        }
        defer f1.Close()
        session.SendInteractionResponse(context.Background(), evt, &disgord.CreateInteractionResponse{
            Type: disgord.InteractionCallbackChannelMessageWithSource,
            Data: &disgord.CreateInteractionResponseData{
                Content: "This is my favourite image",
                Files: []disgord.CreateMessageFile{
                    {f1, "myfavouriteimage.jpg", false},
                },
                Embeds: []*disgord.Embed{{
                    Description: "Look here!",
                    Image: &disgord.EmbedImage{
                        URL: "attachment://myfavouriteimage.jpg",
                    },
                }},
            },
        })
    }
}
func main() {
    client := disgord.New(disgord.Config{
        RejectEvents: []string{disgord.EvtPresenceUpdate, disgord.EvtTypingStart},
        BotToken:     "",
    })
    defer client.Gateway().StayConnectedUntilInterrupted()
    client.Gateway().InteractionCreate(interaction)
}
```

### CODE REVIEW

**Comment by andersfylling** [comment]
Nice! Could you add this as an example, under `/examples`?

File: examples/upload-images-application-command/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module upload-images-application-command
+
+go 1.16
+
+require github.com/andersfylling/disgord v0.32.3 // indirect
```
Comment (andersfylling) [inline examples/upload-images-application-command/go.mod:5]: can you setup a replace directive, like the other examples. Then the CI will detect if a PR breaks any functionality

**Comment by andersfylling** [comment]
does this deprecated `InteractionResponse`?

**Comment by Acnologla** [comment]
> does this deprecate `InteractionResponse`?

Well, it can still be used to answer simpler interactions (which don't need files) to have a small performance gain

**Comment by andersfylling** [comment]
I don't see the performance gain. Given this is an IO call, and that sending files is optional, it seems to me these can be merged?

To avoid confusion as to why there are multiple InteractionResponse objects.
Is there a reason you had to make a new struct?

**Comment by Acnologla** [comment]
> I don't see the performance gain. Given this is an IO call, and that sending files is optional, it seems to me these can be merged?
> 
> To avoid confusion as to why there are multiple InteractionResponse objects. Is there a reason you had to make a new struct?

There's no reason. So I will remove these structures

**Comment by andersfylling** [comment]
I've been a bit occupied. Looking at this again, I see that "attachment" is replaced by "files", yet the discord docs uses attachment?

I think the difference in naming can be confusing, what if I just want to use the attachments field?

**Comment by Acnologla** [comment]
> I've been a bit occupied. Looking at this again, I see that "attachment" is replaced by "files", yet the discord docs uses attachment?
> 
> I think the difference in naming can be confusing, what if I just want to use the attachments field?

CreateInteractionResponse follows the same logic as CreateMessage which does not have the attachment field either

**Comment by andersfylling** [comment]
Ah I see!
Sorry for the delay. 

I think we should update this to support the Attachment field. In this PR, I say we add it so people can take advantage of it.

The CreateMessage struct should also be updated, but that is not related to the scope of this PR.

**Review by andersfylling — APPROVED** [review]

**Pull request merged**